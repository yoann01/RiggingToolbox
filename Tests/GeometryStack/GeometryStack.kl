
require RiggingToolbox;

operator entry(){
    
  GeometryStack stack();
  PolygonMeshSphereGenerator sphereGenerator(2.0, 1, true, true);
  PushModifier pushModifier(3.0);
  stack.addGeometryOperator(sphereGenerator);
  stack.addGeometryOperator(pushModifier);

  EvalContext context();
  {
    GeometrySet geomSet = stack.evaluate(context);

    report("NumGeometries:" + geomSet.size);
    for(Integer i=0; i<geomSet.size; i++){
      report(geomSet.get(i).getDesc(true, true));
    }
  }
  
  {
    GeometrySet geomSet = stack.evaluate(context);

    report("NumGeometries:" + geomSet.size);
    for(Integer i=0; i<geomSet.size; i++){
      report(geomSet.get(i).getDesc(true, true));
    }
  }
}

