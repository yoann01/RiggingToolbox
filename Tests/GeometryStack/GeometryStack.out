NumGeometries:1
Mesh: pointCount: 12 polygonCount: 15 nbAttributeVectors: 33
  Points (adjacent polygons as 'polygon.neighborPolygonIndex',
          borders as '|', closed wing starts as '<<',
          unshared attribute as value@attributeIndex ):
    0: 5 polygons:  <<4.0, 3.0, 2.0, 1.0, 0.0
      Attr positions:{x:+0.0,y:+5.0,z:+0.0}
      Attr normals:{x:+0.0,y:+1.0,z:+0.0}
      Attr uvs0:{x:+0.5,y:+0.0}
    1: 4 polygons:  <<9.1, 4.1, 0.2, 5.0
      Attr positions:{x:+4.330078,y:+2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:+0.5,z:+0.0}
      Attr uvs0:{x:+0.0,y:+0.333313}@17 {x:+0.0,y:+0.0}@15 {x:+1.0,y:+0.333313}@1 {x:+1.0,y:+0.333313}@1
    2: 4 polygons:  <<6.0, 5.1, 0.1, 1.2
      Attr positions:{x:+1.338134,y:+2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:+0.823608}
      Attr uvs0:{x:+0.800048,y:+0.333313}@12 {x:+0.800048,y:+0.333313}@12 {x:+0.800048,y:+0.0}@2 {x:+0.800048,y:+0.333313}@12
    3: 4 polygons:  <<7.0, 6.1, 1.1, 2.2
      Attr positions:{x:-3.502929,y:+2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:+0.509033}
      Attr uvs0:{x:+0.599975,y:+0.333313}@13 {x:+0.599975,y:+0.333313}@13 {x:+0.599975,y:+0.0}@3 {x:+0.599975,y:+0.333313}@13
    4: 4 polygons:  <<8.0, 7.1, 2.1, 3.2
      Attr positions:{x:-3.502929,y:+2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:-0.509033}
      Attr uvs0:{x:+0.400024,y:+0.333313}@14 {x:+0.400024,y:+0.333313}@14 {x:+0.400024,y:+0.0}@4 {x:+0.400024,y:+0.333313}@14
    5: 4 polygons:  <<9.0, 8.1, 3.1, 4.2
      Attr positions:{x:+1.338134,y:+2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:-0.823608}
      Attr uvs0:{x:+0.200012,y:+0.333313}@16 {x:+0.200012,y:+0.333313}@16 {x:+0.200012,y:+0.0}@5 {x:+0.200012,y:+0.333313}@16
    6: 4 polygons:  <<14.2, 9.2, 5.3, 10.1
      Attr positions:{x:+4.330078,y:-2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:-0.5,z:+0.0}
      Attr uvs0:{x:+0.5,y:+1.0}@32 {x:+0.0,y:+0.666626}@18 {x:+1.0,y:+0.666626}@6 {x:+0.800048,y:+0.666626}@19
    7: 4 polygons:  <<11.1, 10.2, 5.2, 6.3
      Attr positions:{x:+1.338134,y:-2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:+0.823608}
      Attr uvs0:{x:+0.599975,y:+0.666626}@22 {x:+0.5,y:+1.0}@20 {x:+0.800048,y:+0.666626}@7 {x:+0.800048,y:+0.666626}@7
    8: 4 polygons:  <<12.1, 11.2, 6.2, 7.3
      Attr positions:{x:-3.502929,y:-2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:+0.509033}
      Attr uvs0:{x:+0.400024,y:+0.666626}@25 {x:+0.5,y:+1.0}@23 {x:+0.599975,y:+0.666626}@8 {x:+0.599975,y:+0.666626}@8
    9: 4 polygons:  <<13.1, 12.2, 7.2, 8.3
      Attr positions:{x:-3.502929,y:-2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:-0.509033}
      Attr uvs0:{x:+0.200012,y:+0.666626}@28 {x:+0.5,y:+1.0}@26 {x:+0.400024,y:+0.666626}@9 {x:+0.400024,y:+0.666626}@9
    10: 4 polygons:  <<14.1, 13.2, 8.2, 9.3
      Attr positions:{x:+1.338134,y:-2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:-0.823608}
      Attr uvs0:{x:+0.0,y:+0.666626}@31 {x:+0.5,y:+1.0}@29 {x:+0.200012,y:+0.666626}@10 {x:+0.200012,y:+0.666626}@10
    11: 5 polygons:  <<14.0, 10.0, 11.0, 12.0, 13.0
      Attr positions:{x:+0.0,y:-5.0,z:+0.0}
      Attr normals:{x:+0.0,y:-1.0,z:+0.0}
      Attr uvs0:{x:+0.200012,y:+0.666626}@30 {x:+1.0,y:+0.666626}@11 {x:+0.800048,y:+0.666626}@21 {x:+0.599975,y:+0.666626}@24 {x:+0.400024,y:+0.666626}@27
  Polygons (connected points as 'point.polygonPointIndex', borders as '|'):
    0: 3 points: 0.4, 2.2, 1.2
    1: 3 points: 0.3, 3.2, 2.3
    2: 3 points: 0.2, 4.2, 3.3
    3: 3 points: 0.1, 5.2, 4.3
    4: 3 points: 0.0, 1.1, 5.3
    5: 4 points: 1.3, 2.1, 7.2, 6.2
    6: 4 points: 2.0, 3.1, 8.2, 7.3
    7: 4 points: 3.0, 4.1, 9.2, 8.3
    8: 4 points: 4.0, 5.1, 10.2, 9.3
    9: 4 points: 5.0, 1.0, 6.1, 10.3
    10: 3 points: 11.1, 6.3, 7.1
    11: 3 points: 11.2, 7.0, 8.1
    12: 3 points: 11.3, 8.0, 9.1
    13: 3 points: 11.4, 9.0, 10.1
    14: 3 points: 11.0, 10.0, 6.0

NumGeometries:1
Mesh: pointCount: 12 polygonCount: 15 nbAttributeVectors: 33
  Points (adjacent polygons as 'polygon.neighborPolygonIndex',
          borders as '|', closed wing starts as '<<',
          unshared attribute as value@attributeIndex ):
    0: 5 polygons:  <<4.0, 3.0, 2.0, 1.0, 0.0
      Attr positions:{x:+0.0,y:+5.0,z:+0.0}
      Attr normals:{x:+0.0,y:+1.0,z:+0.0}
      Attr uvs0:{x:+0.5,y:+0.0}
    1: 4 polygons:  <<9.1, 4.1, 0.2, 5.0
      Attr positions:{x:+4.330078,y:+2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:+0.5,z:+0.0}
      Attr uvs0:{x:+0.0,y:+0.333313}@17 {x:+0.0,y:+0.0}@15 {x:+1.0,y:+0.333313}@1 {x:+1.0,y:+0.333313}@1
    2: 4 polygons:  <<6.0, 5.1, 0.1, 1.2
      Attr positions:{x:+1.338134,y:+2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:+0.823608}
      Attr uvs0:{x:+0.800048,y:+0.333313}@12 {x:+0.800048,y:+0.333313}@12 {x:+0.800048,y:+0.0}@2 {x:+0.800048,y:+0.333313}@12
    3: 4 polygons:  <<7.0, 6.1, 1.1, 2.2
      Attr positions:{x:-3.502929,y:+2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:+0.509033}
      Attr uvs0:{x:+0.599975,y:+0.333313}@13 {x:+0.599975,y:+0.333313}@13 {x:+0.599975,y:+0.0}@3 {x:+0.599975,y:+0.333313}@13
    4: 4 polygons:  <<8.0, 7.1, 2.1, 3.2
      Attr positions:{x:-3.502929,y:+2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:+0.5,z:-0.509033}
      Attr uvs0:{x:+0.400024,y:+0.333313}@14 {x:+0.400024,y:+0.333313}@14 {x:+0.400024,y:+0.0}@4 {x:+0.400024,y:+0.333313}@14
    5: 4 polygons:  <<9.0, 8.1, 3.1, 4.2
      Attr positions:{x:+1.338134,y:+2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:+0.5,z:-0.823608}
      Attr uvs0:{x:+0.200012,y:+0.333313}@16 {x:+0.200012,y:+0.333313}@16 {x:+0.200012,y:+0.0}@5 {x:+0.200012,y:+0.333313}@16
    6: 4 polygons:  <<14.2, 9.2, 5.3, 10.1
      Attr positions:{x:+4.330078,y:-2.5,z:+0.0}
      Attr normals:{x:+0.865966,y:-0.5,z:+0.0}
      Attr uvs0:{x:+0.5,y:+1.0}@32 {x:+0.0,y:+0.666626}@18 {x:+1.0,y:+0.666626}@6 {x:+0.800048,y:+0.666626}@19
    7: 4 polygons:  <<11.1, 10.2, 5.2, 6.3
      Attr positions:{x:+1.338134,y:-2.5,z:+4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:+0.823608}
      Attr uvs0:{x:+0.599975,y:+0.666626}@22 {x:+0.5,y:+1.0}@20 {x:+0.800048,y:+0.666626}@7 {x:+0.800048,y:+0.666626}@7
    8: 4 polygons:  <<12.1, 11.2, 6.2, 7.3
      Attr positions:{x:-3.502929,y:-2.5,z:+2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:+0.509033}
      Attr uvs0:{x:+0.400024,y:+0.666626}@25 {x:+0.5,y:+1.0}@23 {x:+0.599975,y:+0.666626}@8 {x:+0.599975,y:+0.666626}@8
    9: 4 polygons:  <<13.1, 12.2, 7.2, 8.3
      Attr positions:{x:-3.502929,y:-2.5,z:-2.54541}
      Attr normals:{x:-0.700683,y:-0.5,z:-0.509033}
      Attr uvs0:{x:+0.200012,y:+0.666626}@28 {x:+0.5,y:+1.0}@26 {x:+0.400024,y:+0.666626}@9 {x:+0.400024,y:+0.666626}@9
    10: 4 polygons:  <<14.1, 13.2, 8.2, 9.3
      Attr positions:{x:+1.338134,y:-2.5,z:-4.118164}
      Attr normals:{x:+0.267639,y:-0.5,z:-0.823608}
      Attr uvs0:{x:+0.0,y:+0.666626}@31 {x:+0.5,y:+1.0}@29 {x:+0.200012,y:+0.666626}@10 {x:+0.200012,y:+0.666626}@10
    11: 5 polygons:  <<14.0, 10.0, 11.0, 12.0, 13.0
      Attr positions:{x:+0.0,y:-5.0,z:+0.0}
      Attr normals:{x:+0.0,y:-1.0,z:+0.0}
      Attr uvs0:{x:+0.200012,y:+0.666626}@30 {x:+1.0,y:+0.666626}@11 {x:+0.800048,y:+0.666626}@21 {x:+0.599975,y:+0.666626}@24 {x:+0.400024,y:+0.666626}@27
  Polygons (connected points as 'point.polygonPointIndex', borders as '|'):
    0: 3 points: 0.4, 2.2, 1.2
    1: 3 points: 0.3, 3.2, 2.3
    2: 3 points: 0.2, 4.2, 3.3
    3: 3 points: 0.1, 5.2, 4.3
    4: 3 points: 0.0, 1.1, 5.3
    5: 4 points: 1.3, 2.1, 7.2, 6.2
    6: 4 points: 2.0, 3.1, 8.2, 7.3
    7: 4 points: 3.0, 4.1, 9.2, 8.3
    8: 4 points: 4.0, 5.1, 10.2, 9.3
    9: 4 points: 5.0, 1.0, 6.1, 10.3
    10: 3 points: 11.1, 6.3, 7.1
    11: 3 points: 11.2, 7.0, 8.1
    12: 3 points: 11.3, 8.0, 9.1
    13: 3 points: 11.4, 9.0, 10.1
    14: 3 points: 11.0, 10.0, 6.0

